src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -257,23 +273,35 @@ public function putOffExpiration(Key $key, float $ttl)
      */
     public function delete(Key $key)
     {
-        $this->getCollection()->deleteOne([ // filter
-            '_id' => (string) $key,
-            'token' => $this->getUniqueToken($key),
-        ]);
+        $write = new BulkWrite();
+        $write->delete(
+            [ // filter
+                '_id' => (string) $key,
+                'token' => $this->getUniqueToken($key),
```

Comment:
Out of curiosity, with the ID already uniquely identifying a record, what's the reason for adding a condition on `token` as well?

Reply:
The uniqueness of `_id` ensure only 1 process can acquire this lock. The random token is used to identify which process owns the lock. It's necessary here to ensure we don't release a lock owned by an other process.

Reply:
Ah, I see. Thanks for explaining!